### ISSUE TITLE
Configure the HttpClient in the Model

### ISSUE THREAD

**Issue by masterT** [DavidDuwaer/Coloquent#91]
Is there a way to configure the `HttpClient` to use for a `Model`?

I would like to use a specific `Axios` instance that uses [custom interceptors](https://github.com/axios/axios#interceptors). 

The `AxiosHttpClient` is not exposed, but maybe it should be.

Maybe something like:

```ts
import customAxiosInstance from './path/to/axios/custom/instance'
import { Model, HttpClient, AxiosHttpClient } from 'coloquent'

export default abstract class ApplicationModel extends Model {
  public getHttpClient(): HttpClient {
    return new AxiosHttpClient(Api)
  }
  // ...
}
```

This might be related to https://github.com/DavidDuwaer/Coloquent/issues/37.

**Comment by DavidDuwaer** [comment]
If that's going to work, then that method should be static. What happens if you try that?

**Comment by masterT** [comment]
Oh, I did not realized that this is already implemented, but it can't be define in an abstract class (ex: ApplicationModel). It would be nice to be able to set it once in an ApplicationModel, so every Model that extends it would use the same HttpClient.

**Comment by DavidDuwaer** [comment]
Yeah I agree. The fact that `getHttpClient` is static right now makes "overriding" it unpredictable/confusing at best and impossible at worst. I see the only place where `.getHttpClient()` gets called is in Builder:46. In that place, an instance of the relevant `Model` is also available, so `.getHttpClient()` could be nonstatic. If it were nonstatic, it could be overridden like normal methods and resolve this issue.

**Comment by masterT** [comment]
If it is static then, I could be used like this.

```ts
abstract class Model {
  protected static httpClient: HttpClient;

  public static getHttpClient(): HttpClient {
      if (!this.httpClient) {
          this.httpClient = new AxiosHttpClient();
      }
      return this.httpClient
  }
}

abstract class BaseModel extends Model {
  protected static httpClient: HttpClient = new AxiosHttpClient();
}
```

This means changing the `httpClient` property from `private` to `protected`. Maybe the getter could be removed too, in this case:

```ts
abstract class Model {
  protected static httpClient: HttpClient = new AxiosHttpClient();
}
```

**Comment by masterT** [comment]
I think the best would be to define `httpClient` as `abstract static httpClient: HttpClient`, but this seems not possible in TypeScript.

**Comment by masterT** [comment]
It feels like setting the base URL should not be done in the `HttpClient`, maybe it should only receive a full URL, and add a method in `Model` to expose the full URL, and a method on a `Model` instance to expose to full URL for that resource.

**Comment by masterT** [comment]
The PR haa been merged, it's available with the version `3.0.1-beta`.
